aggregator/handler/router/with_ack.go

```diff
@@ -47,6 +47,7 @@ type message struct {
 	buffer *common.RefCountedBuffer
 }
 
+// TODO: Pool the messages if needed.
```

Comment:
nit: can you assign this to you (i.e., making it `TODO (cw9)`)? I usually prefer to put a name next to the TODOs or otherwise the ownership of TODOs is not really clear when I clean them up later.